genai-perf/genai_perf/inputs/output_format_converter.py

```diff
@@ -0,0 +1,780 @@
+        text_input_headers: List[str],
+        content: str,
+    ) -> str:
+        new_text_input = ""
+
+        if (
+            header in system_role_headers
+            or header in user_role_headers
+            or header in text_input_headers
+        ):
+            new_text_input = content
+
+        return new_text_input
+
+    def _add_message_to_json(self, openai_json: Dict, message: Optional[Dict]) -> Dict:
+        if message:
+            openai_json["payload"][0]["messages"].append(message)
+
+        return openai_json
+
+    def _add_optional_tags_to_openai_json(
+        self,
+        openai_json: Dict,
+        config: InputsConfig,
+        model_name: str = "",
+    ) -> None:
+        payload = openai_json["payload"][0]
+        payload["model"] = model_name
+        if config.add_stream:
+            payload["stream"] = True
+        if config.output_tokens_mean != DEFAULT_OUTPUT_TOKENS_MEAN:
+            payload["max_tokens"] = int(
+                random.gauss(config.output_tokens_mean, config.output_tokens_stddev)
+            )
+        for key, value in config.extra_inputs.items():
+            payload[key] = value
+
+
+class OpenAICompletionsConverter(BaseConverter):
+    def convert(
+        self,
+        generic_dataset: Dict,
+        config: InputsConfig,
+    ) -> Dict:
+        (
+            system_role_headers,
+            user_role_headers,
+            text_input_headers,
+        ) = self._determine_json_feature_roles(generic_dataset)
+
+        pa_json = self._populate_openai_completions_output_json(
+            generic_dataset,
+            system_role_headers,
+            user_role_headers,
+            text_input_headers,
+            config,
+        )
+
+        return pa_json
+
+    def _populate_openai_completions_output_json(
+        self,
+        generic_dataset: Dict,
+        system_role_headers: List[str],
+        user_role_headers: List[str],
+        text_input_headers: List[str],
+        config: InputsConfig,
+    ) -> Dict:
+        pa_json = deepcopy(EMPTY_JSON_IN_OPENAI_PA_FORMAT)
+
+        for index, entry in enumerate(generic_dataset["rows"]):
+            iter_model_name = self._select_model_name(config, index)
+            pa_json["data"].append({"payload": []})
+            pa_json["data"][index]["payload"].append({"prompt": ""})
+
+            for header, content in entry.items():
+                new_prompt = self._create_new_prompt(
+                    header,
+                    system_role_headers,
+                    user_role_headers,
+                    text_input_headers,
+                    content,
+                )
+
+                pa_json = self._add_new_prompt_to_json(pa_json, index, new_prompt)
+
+            self._add_optional_tags_to_openai_json(
+                pa_json["data"][index],
+                config,
+                iter_model_name,
+            )
+
+        return pa_json
+
+    def _create_new_prompt(
```

Comment:
Same comment as previous - should avoid duplication of code

Reply:
Done!